docs/develop/guides/working-with-momento-tokens.md

```diff
@@ -0,0 +1,131 @@
+<Tabs>
+    <TabItem value="nodejs" label="Node.js" default>
+
+```javascript
+const fs = require('fs');
+const { AuthClient, GenerateAuthToken, RefreshAuthToken, ExpiresAt, ExpiresIn } = require('@gomomento/sdk');
+const dotenv = require('dotenv');
+
+// Value must contain a valid go momento endpoint
+const controlEndpoint = "valid-gomomento-endpoint";
+// Value must contain a valid momento auth token
+const authToken = "valid-momento-auth-token";
+
+dotenv.config();
+
+// Creates the Momento auth client object
+async function createAuthClient() {
+  return new AuthClient();
+}
+
+// Creates the Momento cache client object
+async function createCacheClient(momentoAuthToken: string) {
+  return new CacheClient({
+    configuration: Configurations.Laptop.v1(),
+    credentialProvider: CredentialProvider.fromString({
+      environmentVariableName: momentoAuthToken,
+    }),
+    defaultTtlSeconds: 600,
+  });
+}
+
+async function refreshExpiringToken(authToken: string, refreshToken: string) {
+  const refreshTokenResponse = await authClient.refreshAuthToken(
+    CredentialProvider.fromString({
+      authToken: authToken,
+    }),
+    refreshToken
+  );
+
+  if (refreshTokenResponse instanceof RefreshAuthToken.Error) {
+    throw new Error(`Error: ${refreshTokenResponse.message()}`);
+  }
+  console.log('Auth token successfully refreshed!');
+  
+  return refreshTokenResponse;
+}
+
+async function usingAuthToken(cacheClient: CacheClient) {
+  const result = await cacheClient.listCaches();
+  if (result instanceof ListCaches.Success) {
+    console.log(
+      `Caches:\n${result
+        .getCaches()
+        .map(c => c.getName())
+        .join('\n')}\n\n`
+    );
+  } else if (result instanceof ListCaches.Error) {
+    throw new Error(
+      `An error occurred while attempting to list caches: ${result.errorCode()}: ${result.toString()}`
+    );
+  }
+}
+
+async function run() {
+  const byteArray = await readFile(filePath);
+  if (byteArray === null) {
+    return;
+  }
+
+  const authClient = await createAuthClient();
+
+  const cacheClient = await createCacheClient(authToken);
+  
+  await usingAuthToken(cacheClient);
+  
+  const refreshTokenResponse = await refreshExpiringToken(authToken, refreshToken);
+}
+
+run();
+```
+## Step 4: Automating Token Refresh
+Once your application is running in production. You will likely need an automated script that periodically refresh your auth token, before it expires. If you are running in AWS, an easy way to do this is by simply scheduling a lambda that does this for you. Check out our [1-click deploy example lambda](link to github) that you can use to automatically refresh your token in AWS Secrets Manager. 
+
+Remember, while the lambda (or your own automated script) refreshes the auth token, your application also needs to check AWS Secrets Manager (or wherever you are storing your tokens) periodically for newly refreshed tokens!
```

Comment:
maybe mention that it's a good idea to also store the expiration date as part of your secret so that it's easy for the client to know when the token will expire?